File: python/ejemplo_08.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import gi
try:
    gi.require_version('Gtk', '3.0')
except ValueError:
    exit(1)
from gi.repository import Gtk


if __name__ == '__main__':
    dialog = Gtk.FileChooserDialog()
    dialog.set_title('https://www.atareao.es')
    dialog.add_button('Si', Gtk.ResponseType.YES)
    dialog.add_button('No', Gtk.ResponseType.NO)
    try:
        if dialog.run() == Gtk.ResponseType.YES:
            print(dialog.get_filename())
            exit(0)
    except KeyboardInterrupt:
        pass
    exit(1)

File: javascript/ejemplo_08.js
#!/usr/bin/gjs

imports.gi.versions.Gtk = '3.0';
const {Gtk} = imports.gi;

Gtk.init(null);

function show_dialog(){
    let filter = new Gtk.FileFilter();
    filter.add_mime_type('text/plain');
    let dialog = new Gtk.FileChooserDialog({
        title: 'https://www.atareao.es',
        action: Gtk.FileChooserAction.OPEN,
        filter: filter,
        select_multiple: false
        });
    dialog.add_button('Si', Gtk.ResponseType.YES);
    dialog.add_button('No', Gtk.ResponseType.NO);
    try{
        if(dialog.run() == Gtk.ResponseType.YES){
            print(dialog.get_filename());
            return 0;
        }
    }catch(error){
    }
    return 1;
}
show_dialog();
